pandas/core/reshape/reshape.py

```diff
@@ -863,6 +863,8 @@ def get_dummies(
         # determine columns being encoded
         if columns is None:
             data_to_encode = data.select_dtypes(include=dtypes_to_encode)
+        elif not is_list_like(columns):
+            raise TypeError("Input must be a list-like of list-likes")
```

Comment:
The message should state which parameter is incorrect (columns?)

And is it supposed to be a  list-like of list-likes? This block is just checking that it's a sequence, but not making any assertion about what each element is.

Reply:
> The message should state which parameter is incorrect
I have updated the Error message based on your comment but let me point out [this code](https://github.com/pandas-dev/pandas/blob/master/pandas/core/reshape/util.py#L32) which was the reason I did not update the error message.  

>making any assertion about what each element is.
I think this is out of scope because for all the functions definitions (involving list-like object as parameters ) I referred to in pandas, none of them had this check. @jbrockmendel Is this something which should have been checked?